confluent_kafka_helpers/consumer.py

```diff
@@ -123,16 +134,47 @@ def _message_generator(self):
                 continue
 
             statsd.increment(f'{base_metric}.consumer.message.count.total')
-            message = Message(message)
 
-            with tracer.extract_headers_and_start_span(
-                operation_name='kafka.consumer.consume', headers=message._meta.headers
+            value, resource_name = message.value(), message.topic()
+            message_class = value.get("class") if isinstance(value, dict) else None
```

Comment:
Introspect the message and extract the `class` (name of the event/command) and use it together with the topic to set the resource name, eg: `inventory.article.commands:UpdateArticle`. This is the equivalent of `POST /articles/<article_id>` of an API.

Reply:
Having this information, we can use the new "Traces" view in Datadog APM to query eg. services that produce/consume a specific event/command